### PR TITLE
Abi call refactors integration

### PR DESCRIPTION
**Pull request by julia-zack**
There is an ongoing effort to organize and refactor Bridge classes. One of the goals is to create specific packages to hold related classes, having in mind to avoid circular dependencies.

One of the classes we have identified that could be moved to their package are those related to ABI voting (ABICallElection, ABICallSpec, ABICallVoteResult and AddressBasedAuthorizer).

This PR includes creating a new `/vote` package with all those classes, some refactors, some code smells fixes, some logs addition and some new tests addition when needed.

### CODE REVIEW

**Comment by marcos-iov** [comment]
pipeline:run

File: rskj-core/src/main/java/co/rsk/peg/vote/ABICallElection.java [hunk 1]
```diff
@@ -59,47 +69,57 @@ public void clear() {
...
+    }
+
+    private boolean areEnoughVotes(int votesSize) {
+        return votesSize >= authorizer.getRequiredAuthorizedKeys();
     }
 
     /**
      * Removes the entry votes for the current winner of the election
      */
     public void clearWinners() {
```
Comment (nathanieliov) [inline rskj-core/src/main/java/co/rsk/peg/vote/ABICallElection.java:119]: Should it not be `clearWinner`, in singular?

**Review by nathanieliov — COMMENTED** [review]
LGTM. Just a comment.

**Review by jeremy-then — APPROVED** [review]
LGTM.

**Review by marcos-iov — APPROVED** [review]

**Review by josedahlquist — APPROVED** [review]

**Pull request merged**